activation/post_verifier.go

```diff
@@ -188,8 +193,10 @@ func (v *OffloadingPostVerifier) scale(target int) {
 		v.workers = toKeep
 		for _, worker := range toStop {
 			close(worker.stop)
+			stopped = append(stopped, worker.stopped)
 		}
```

Comment:
Why is the caller responsible to await the change? I think we could just leave `scale()` synchronous and await the call here:

```suggestion
		for _, worker := range toStop {
			close(worker.stop)
		}
		for _, worker := range toStop {
			<-worker.stopped
		}
```

So if `scale` returns it has finished scaling.

Reply:
The caller is not responsible for awaiting, it's optional. The `scale()` function doesn't wait for workers to stop because there is no reason for it. In most situations, it's enough to trigger workers to stop. I only need to wait for workers to stop in tests.

Reply:
Are you sure that this shouldn't be the case? `run` does not await the call, meaning if 2 events arrive while scaling still happens they will interfere with each other?

Reply:
It doesn't matter if `scale()` is called while the workers are still closing. Scaling up will create new instances. It only matters that `scale()` is not called from many threads (it isn't).

Anyway, I can make `scale()` blocking (wait for workers to finish) as it will simplify the code and it doesn't matter anyway.